server/src/main/java/org/elasticsearch/action/search/TransportMultiSearchAction.java

```diff
@@ -159,17 +164,35 @@ public void onFailure(final Exception e) {
 
             private void handleResponse(final int responseSlot, final MultiSearchResponse.Item item) {
                 responses.set(responseSlot, item);
+                
+                if (recallGoal > 0 ) {
+                    SearchHits hits = item.getResponse().getHits();
+                    if (hits != null) {
+                        long recall = hits.getTotalHits().value;
+                        if (recall >= recallGoal) {                    
```

Comment:
Should this be cumulative? ie, your recall goal is 10, you get four from the first query and seven from the second, and so the combination of the two is enough to avoid running the third query.

Reply:
I don't think cumulative is helpful because it's not just about gathering hits.
A strict query might be certain of the matching subject matter and therefore want to use terms aggregations whereas a sloppy query might produce some good hits up top but a very long tail of junk so a sampler aggregation and significant_terms aggs would be more appropriate to return only [relevant facets](https://youtu.be/W8DJYfAKKLA).